runtime/compiler/env/VMJ9.cpp

```diff
@@ -8524,6 +8541,12 @@ TR_J9SharedCacheVM::isClassVisible(TR_OpaqueClassBlock * sourceClass, TR_OpaqueC
           TR_J9VMBase::isClassVisible(sourceClass, destClass);
    }
 
+bool
```

Comment:
Is there some inherent reason this optimization cannot be done under AOT or is it just something you are not handling at this point in time ?

Reply:
Oh you're viewing this commit by commit; in 5e6be87f7e9f4847742cbb2ebc8b8442b669caa4 (Add AOT Validations) is where I actually add the necessary validations to make sure this works under AOT